homeassistant/components/onvif/device.py

```
@@ -565,6 +590,46 @@ async def async_perform_ptz(
                     "Zoom": {"x": speed_val},
                 }
                 await ptz_service.RelativeMove(req)
+
+            elif move_mode == EMULATE_RELATIVE_MOVE:
+                # Using continous move, with a set time for an step
+                # Guard against unsupported operation
+                try:
+                    req = ptz_service.create_type(CONTINUOUS_MOVE)
+                    req.ProfileToken = profile.token
+                    if not profile.ptz or not profile.ptz.continuous:
+                        LOGGER.warning(
+                            "ContinuousMove not supported on device '%s'", self.name
+                        )
+                        return
+
+                    # only works if one movement is set, with multiple there is no reponse at least with Arenti camera
+
+                    velocity = {}
+                    if pan is not None or tilt is not None:
+                        velocity["PanTilt"] = {"x": pan_val, "y": tilt_val}
+
+                    # maybe add a loop to accept two movements
+                    if pan is not None and tilt is not None:
+                        LOGGER.warning("ContinuousMove Not supporting two movements")
+
+                    if zoom is not None:
+                        velocity["Zoom"] = {"x": zoom_val}
+
+                    req.Velocity = velocity
+                    time_to_move = calculate_sleep_time(distance, full_rotation_time= 8)
+                    await ptz_service.ContinuousMove(req)
+                    await asyncio.sleep(time_to_move)
+                    velocity = {"PanTilt": {"x": 0, "y": 0}, "Zoom": {"x": 0}}
+                    req.Velocity = velocity
+                    await ptz_service.ContinuousMove(req)
+
+                except:
```

same here, can you use the exception you expect rather than a bare except?